javalib/src/main/scala/java/util/concurrent/locks/ReentrantLock.scala

```diff
@@ -0,0 +1,48 @@
+package java.util.concurrent.locks
+
+import java.io.Serializable
+import java.lang.Thread
+import java.util.concurrent.TimeUnit
+
+class ReentrantLock(fair: Boolean) extends Lock with Serializable {
+
+  private var locked = 0
+
+  def this() = this(false)
+
+  def lock(): Unit = locked += 1
+
+  def lockInterruptibly(): Unit =
+    if (Thread.currentThread().isInterrupted()) throw new InterruptedException()
```

Comment:
The spec says that in case it's interrupted, "`InterruptedException` is thrown and *the current thread's interrupted status is cleared*.". So you should use `Thread.interrupted()` instead of `Thread.currentThread().isInterrupted()`.

Btw, an if/else in statement position should always have its branches on dedicated lines.